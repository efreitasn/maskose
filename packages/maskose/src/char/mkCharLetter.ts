import { MaskCharPredicateFn } from '.';

export const MASK_CHAR_LETTER_TYPE = 'MASK_CHAR_LETTER_TYPE';

export type MaskCharLetterObj = {
  type: typeof MASK_CHAR_LETTER_TYPE;
  predicateFn?: MaskCharPredicateFn;
};

/**
 * A character in the range A to Z (case insensitive) expected to be in the value to be masked
 * @param predicateFn Function that will decide if the character will be present in the mask
 */
export default function mkCharLetter(predicateFn?: MaskCharPredicateFn): MaskCharLetterObj {
  return {
    type: MASK_CHAR_LETTER_TYPE,
    predicateFn
  };
}