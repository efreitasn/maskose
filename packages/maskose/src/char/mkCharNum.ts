import { MaskCharPredicateFn } from '.';

export const MASK_CHAR_NUM_TYPE = 'MASK_CHAR_NUM_TYPE';

export type MaskCharNumObj = {
  type: typeof MASK_CHAR_NUM_TYPE;
  predicateFn?: MaskCharPredicateFn;
};

/**
 * A character in the range 0 to 9 expected to be in the value to be masked
 * @param predicateFn Function that will decide if the character will be present in the mask
 */
export default function mkCharNum(predicateFn?: MaskCharPredicateFn): MaskCharNumObj {
  return {
    type: MASK_CHAR_NUM_TYPE,
    predicateFn
  };
}