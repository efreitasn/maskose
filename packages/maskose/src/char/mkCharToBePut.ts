import { MaskCharPredicateFn } from '.';

const MASK_CHAR_TO_BE_PUT_TYPE = 'MASK_CHAR_TO_BE_PUT_TYPE';

export type MaskCharToBePutObj = {
  type: typeof MASK_CHAR_TO_BE_PUT_TYPE;
  char: string;
  predicateFn?: MaskCharPredicateFn;
};

/**
 * A character that's expected to be put in the mask when formatting instead of being a character
 * expected to be in the value to be masked. For example, the dash (-) in a phone number mask is
 * such a character
 * @param char Character to be put in the mask
 * @param predicateFn Function that will decide if the character will be present in the mask
 */
export default function mkCharToBePut(char: string, predicateFn?: MaskCharPredicateFn): MaskCharToBePutObj {
  return {
    type: MASK_CHAR_TO_BE_PUT_TYPE,
    char,
    predicateFn
  };
}