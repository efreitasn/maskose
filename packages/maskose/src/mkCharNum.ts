const MASK_CHAR_NUM_TYPE = 'MASK_CHAR_NUM_TYPE';

export type MaskCharPredicateFn = (arg: {
  value: string;
}) => boolean;

export type MkCharNumObj = {
  type: typeof MASK_CHAR_NUM_TYPE;
  predicateFn?: MaskCharPredicateFn;
};

export default function mkCharNum(predicateFn?: MaskCharPredicateFn): MkCharNumObj {
  return {
    type: MASK_CHAR_NUM_TYPE,
    predicateFn
  };
}