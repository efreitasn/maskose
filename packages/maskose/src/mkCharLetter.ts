import { MaskCharPredicateFn } from './mkCharNum';

const MASK_CHAR_LETTER_TYPE = 'MASK_CHAR_LETTER_TYPE';

export type MkCharLetterObj = {
  type: typeof MASK_CHAR_LETTER_TYPE;
  predicateFn?: MaskCharPredicateFn;
};

export default function mkCharLetter(predicateFn?: MaskCharPredicateFn): MkCharLetterObj {
  return {
    type: MASK_CHAR_LETTER_TYPE,
    predicateFn
  };
}