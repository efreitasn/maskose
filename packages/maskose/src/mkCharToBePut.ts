import { MaskCharPredicateFn } from './mkCharNum';

const MASK_CHAR_TO_BE_PUT_TYPE = 'MASK_CHAR_TO_BE_PUT_TYPE';

export default function mkCharToBePut(char: string, predicateFn?: MaskCharPredicateFn) {
  return {
    type: MASK_CHAR_TO_BE_PUT_TYPE,
    char,
    predicateFn
  };
}