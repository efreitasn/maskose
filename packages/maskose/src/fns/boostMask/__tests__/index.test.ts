import mkBoostMask from '..';
import mkCharNum from '../../../mask/chars/num';
import mkCreate from '../../create';
import mkCharToBePut from '../../../mask/chars/toBePut';
import mkCharLetter from '../../../mask/chars/letter';
import { mkMaskBoostEndless, mkMaskBoostRightToLeft } from '../../..';

it('should match the object returned by mkBoostMask()()', () => {
  const mask = mkCreate([
    mkCharNum(),
    mkCharToBePut('-'),
    mkCharLetter()
  ]);
  const result = mkBoostMask(mask)([
    mkMaskBoostEndless(),
    mkMaskBoostRightToLeft()
  ]);

  expect(result).toMatchSnapshot();
});