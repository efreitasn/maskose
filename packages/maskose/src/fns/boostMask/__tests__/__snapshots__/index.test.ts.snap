// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the object returned by mkBoostMask()() 1`] = `
Object {
  "chars": Array [
    Object {
      "maskedValueLengthConditions": Array [],
      "regExp": /\\[0-9\\]/,
      "repetitions": 1,
      "type": "MASKOSE_CHAR_NUM_TYPE",
      "valueToBeMaskedLengthConditions": Array [],
    },
    Object {
      "char": "-",
      "maskedValueLengthConditions": Array [],
      "regExp": /-/,
      "repetitions": 1,
      "type": "MASKOSE_CHAR_TO_BE_PUT_TYPE",
      "valueToBeMaskedLengthConditions": Array [],
    },
    Object {
      "maskedValueLengthConditions": Array [],
      "regExp": /\\[A-Z\\]/i,
      "repetitions": 1,
      "type": "MASKOSE_CHAR_LETTER_TYPE",
      "valueToBeMaskedLengthConditions": Array [],
    },
  ],
  "direction": "right-to-left",
  "endless": true,
}
`;
